twisted/conch/client/default.py

```diff
@@ -156,21 +161,25 @@ def _ebSetAgent(self, f):
         userauth.SSHUserAuthClient.serviceStarted(self)
 
     def _getPassword(self, prompt):
-        try:
-            oldout, oldin = sys.stdout, sys.stdin
-            sys.stdin = sys.stdout = open('/dev/tty','r+')
-            p=getpass.getpass(prompt)
-            sys.stdout,sys.stdin=oldout,oldin
-            return p
-        except (KeyboardInterrupt, IOError):
-            print()
-            raise ConchError('PEBKAC')
+        with self._replaceStdoutStdin():
+            try:
+                p=getpass.getpass(prompt)
+                return p
+            except (KeyboardInterrupt, IOError):
+                print()
+                raise ConchError('PEBKAC')
 
     def getPassword(self, prompt = None):
```

Comment:
The type of `prompt` seems pretty important now; would you mind adding a docstring that specifies its `@type`?  Looks like it's `native L{str}` if I understand properly?